device_backends/Subdevice/src/SubdeviceRegisterAccessor.cc

```diff
@@ -26,41 +26,51 @@ namespace ChimeraTK {
 
   bool SubdeviceRegisterAccessor::doWriteTransfer(ChimeraTK::VersionNumber) {
     std::lock_guard<decltype(_backend->mutex)> lockGuard(_backend->mutex);
-    // This is "_numberOfWords / _accData->getNumberOfSamples()" rounded up:
-    auto nTransfers = (_numberOfWords + _accData->getNumberOfSamples() - 1) / _accData->getNumberOfSamples();
+    size_t nTransfers;
+    if(_backend->hasAreaParam())
+      // for areaHandshake case with 1D array
+      nTransfers = _numberOfWords;
+    else {
+      // This is "_numberOfWords / _accData->getNumberOfSamples()" rounded up:
+      nTransfers = (_numberOfWords + _accDataArea->getNumberOfSamples() - 1) / _accDataArea->getNumberOfSamples();
+    }
     try {
       size_t idx = 0;
       for(size_t adr = _startAddress; adr < _startAddress + nTransfers; ++adr) {
-        if(_backend->type == SubdeviceBackend::Type::threeRegisters) {
+        if(_backend->type == SubdeviceBackend::Type::threeRegisters ||
+            _backend->type == SubdeviceBackend::Type::areaHandshake) {
           size_t retry = 0;
           size_t max_retry = _backend->timeout * 1000 / _backend->sleepTime;
           while(true) {
             _accStatus->read();
             if(_accStatus->accessData(0) == 0) break;
             if(++retry > max_retry) {
               throw ChimeraTK::runtime_error("Write to register '" + _name +
-                  "' failed: timout waiting for cleared busy flag (" + _accStatus->getName() + ")");
+                  "' failed: timeout waiting for cleared busy flag (" + _accStatus->getName() + ")");
             }
             usleep(_backend->sleepTime);
           }
         }
-        else {
+        else { // twoRegisters type
           usleep(_backend->sleepTime);
         }
-        _accAddress->accessData(0) = adr;
-        _accAddress->write();
-        usleep(_backend->addressToDataDelay);
-        for(size_t innerOffset = 0; innerOffset < _accData->getNumberOfSamples(); ++innerOffset) {
-          if(idx < _numberOfWords) {
-            _accData->accessData(0, innerOffset) = _buffer[idx];
-          }
-          else {
+        if(_backend->hasAreaParam()) {
```

Comment:
Even though this seems to have the right behaviour, I would check for the areaHandshake type instead of calling this function. The code in the if case is not executed for the normal area type, hence this might be confusing.